libtransact/src/scheduler/multi/mod.rs

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+
+//! A `Scheduler` which runs multiple sub-schedulers.
+
+mod core;
+mod shared;
+
+use crate::execution::executor::Executor;
+use crate::protocol::batch::BatchPair;
+use crate::scheduler::{
+    BatchExecutionResult, ExecutionTask, ExecutionTaskCompletionNotifier, Scheduler,
+};
+
+use std::sync::mpsc;
+use std::sync::mpsc::Sender;
+use std::sync::{Arc, Mutex};
+
+/// A `Scheduler` implementation which runs multiple sub-schedulers.
+pub struct MultiScheduler {
+    shared_lock: Arc<Mutex<shared::MultiSchedulerShared>>,
+    core_handle: Option<std::thread::JoinHandle<()>>,
+    core_tx: Sender<core::MultiSchedulerCoreMessage>,
+}
+
+impl MultiScheduler {
+    /// Returns a newly created `MultiScheduler` that runs the specified sub-schedulers.
+    pub fn new(
+        mut schedulers: Vec<Box<dyn Scheduler + Send>>,
+        executor: &Executor,
+    ) -> MultiScheduler {
```

Comment:
The external interface of the MultiScheduler is different from the other schedulers; it takes the sub-schedulers as an argument and it requires access to the Executor to execute the sub-schedulers. Also, since it is not really an actual scheduler itself, its `take_task_iterator` and `new_notifer` methods aren't necessary, so I have left them unimplemented. I'm not sure there's a way to provide the same external interface as the other schedulers.